nlp/2021_06_18_data_augmentation/totw_nlp_dat_aug.ipynb

```diff
@@ -0,0 +1,6046 @@
+            "_view_module_version": "1.2.0",
+            "grid_template_areas": null,
+            "object_position": null,
+            "object_fit": null,
+            "grid_auto_columns": null,
+            "margin": null,
+            "display": null,
+            "left": null
+          }
+        }
+      }
+    }
+  },
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "LXro9efVU84k"
+      },
+      "source": [
+        "# Data augmentation techniques for NLP\n",
+        "\n",
+        "The training size will impact the performace of a model heavily, this notebook will discuss the possibilities of performing data augmentation on a NLP dataset. Data augmentation techniques are used to generate additional samples. Data augmentation is already standard practice in computer vision projects, but can also be leveraged in multilingual NLP problems. We'll use a limited trainingset to simulate a real-world use case, where we often are constrained by the size of the available data. We'll focuss on using backtranslation and contextual word-embedding insertions as a data augmentation techniques."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "acwqGIAyU84m"
+      },
+      "source": [
+        "## 🛠️ Getting started\n",
+        "\n",
+        "The cells below will setup everything that is required to get started with data augmentation and finetuning an NLP model with the HuggingFace API."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "buIDjl_wU84m"
+      },
+      "source": [
+        "### Setup"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "v-ItvjqP4Cxn"
+      },
+      "source": [
+        "!pip install -q transformers sentencepiece datasets tokenizers nltk nlpaug "
+      ],
+      "execution_count": 15,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "Ir6zD__FU84n"
+      },
+      "source": [
+        "### Imports"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "hM-09Chsj7wI"
+      },
+      "source": [
+        "import re\n",
+        "import numpy as np\n",
+        "import pandas as pd \n",
+        "\n",
+        "import nltk\n",
+        "import nlpaug.flow as naf\n",
+        "import nlpaug.augmenter.word as naw\n",
+        "import plotly.graph_objects as go\n",
+        "from transformers import pipeline, AutoTokenizer, AutoModelForSequenceClassification, TrainingArguments, Trainer, TrainerCallback\n",
+        "from datasets import load_dataset, concatenate_datasets, load_from_disk, load_metric"
+      ],
+      "execution_count": 16,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "WlCiqdnZU84n"
+      },
+      "source": [
+        "### Download translation pipelines and tokenizer"
```

Comment:
Some extra description why we do this and what's going on

Reply:
Or even better: moving this cell further down to the 'backtranslation' paragraph